hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ARequesterPays.java

```diff
@@ -102,14 +88,30 @@ public void testRequesterPaysDisabledFails() throws Throwable {
     }
   }
 
-  private Path getRequesterPaysPath(Configuration conf) {
-    String requesterPaysFile =
-        conf.getTrimmed(KEY_REQUESTER_PAYS_FILE, DEFAULT_REQUESTER_PAYS_FILE);
-    S3ATestUtils.assume(
-        "Empty test property: " + KEY_REQUESTER_PAYS_FILE,
-        !requesterPaysFile.isEmpty()
-    );
-    return new Path(requesterPaysFile);
+  /**
+   * Use this after creating the file system, as this is when bucket-specific
+   * overrides are applied.
+   * @param conf Hadoop configuration from FS to mutate
+   * @param requesterPaysEnabled Indicate if requester pays be on or off
+   */
+  private static void updateConf(
+      Configuration conf,
+      boolean requesterPaysEnabled
```

Comment:
nit: we tend to pull that ) onto the same line as the last paraem

Reply:
makes sense, will note for future.

i removed this code now since the change was unnecessary and was making the patch longer. (I had some confusion around the bucket conf propagation when i wrote it)